spring-integration-ftp/src/main/java/org/springframework/integration/ftp/session/FtpRemoteFileTemplate.java

```diff
@@ -58,21 +60,33 @@ public T doInSession(Session<FTPFile> session) throws IOException {
 		});
 	}
 
+	/**
+	 * This particular FTP implementation is based on the {@link FTPClient#listNames(String)},
+	 * because not all FTP servers properly implement the {@code STAT} command and
+	 * it doesn't provide correct information for all cases.
+	 * Also, since this method is intended for only file existence checks,
+	 * it doesn't rely on the {@link Session#exists(String)}, because the last one uses
+	 * {@link FTPClient#changeWorkingDirectory(String)} as a fallback option, when
+	 * {@link FTPClient#listNames(String)} may not return the proper answer for directories
+	 * on some servers, too.
+	 * @param path the remote file path to check.
+	 * @return true or false if remote file exists or not.
+	 */
 	@Override
 	public boolean exists(final String path) {
-		return executeWithClient(new ClientCallback<FTPClient, Boolean>() {
+		return doExecuteWithClient(new ClientCallback<FTPClient, Boolean>() {
 
 			@Override
 			public Boolean doWithClient(FTPClient client) {
 				try {
-					return client.getStatus(path) != null;
+					String[] names = client.listNames(path);
+					return !ObjectUtils.isEmpty(names);
```

Comment:
Seems a bit of a "cop-out", since this is a well-known issue.

How about a flag on the template to control the behavior? Default being the current behavior.

In any case (user subclass or a flag), we need a property on the outbound endpoints so the user can inject a customized template.